sdk/resourcemanager/azure-resourcemanager-eventhubs/src/main/java/com/azure/resourcemanager/eventhubs/implementation/EventHubImpl.java

```diff
@@ -0,0 +1,515 @@
+    }
+
+    @Override
+    public EventHubImpl withExistingNamespaceId(String namespaceId) {
+        this.ancestor = new Ancestors().new OneAncestor(selfId(namespaceId));
+        return this;
+    }
+
+    @Override
+    public EventHubImpl withNewStorageAccountForCapturedData(
+        Creatable<StorageAccount> storageAccountCreatable, String containerName) {
+        this.captureSettings.withNewStorageAccountForCapturedData(storageAccountCreatable, containerName);
+        return this;
+    }
+
+    @Override
+    public EventHubImpl withExistingStorageAccountForCapturedData(
+        StorageAccount storageAccount, String containerName) {
+        this.captureSettings.withExistingStorageAccountForCapturedData(storageAccount, containerName);
+        return this;
+    }
+
+    @Override
+    public EventHubImpl withExistingStorageAccountForCapturedData(
+        String storageAccountId, String containerName) {
+        this.captureSettings.withExistingStorageAccountForCapturedData(storageAccountId, containerName);
+        return this;
+    }
+
+    @Override
+    public EventHubImpl withDataCaptureEnabled() {
+        this.captureSettings.withDataCaptureEnabled();
+        return this;
+    }
+
+    @Override
+    public EventHubImpl withDataCaptureDisabled() {
+        this.captureSettings.withDataCaptureDisabled();
+        return this;
+    }
+
+    @Override
+    public EventHubImpl withDataCaptureWindowSizeInSeconds(int sizeInSeconds) {
+        this.captureSettings.withDataCaptureWindowSizeInSeconds(sizeInSeconds);
+        return this;
+    }
+
+    @Override
+    public EventHubImpl withDataCaptureSkipEmptyArchives(Boolean skipEmptyArchives) {
+        this.captureSettings.withDataCaptureSkipEmptyArchives(skipEmptyArchives);
+        return this;
+    }
+
+    @Override
+    public EventHubImpl withDataCaptureWindowSizeInMB(int sizeInMB) {
+        this.captureSettings.withDataCaptureWindowSizeInMB(sizeInMB);
+        return this;
+    }
+
+    @Override
+    public EventHubImpl withDataCaptureFileNameFormat(String format) {
+        this.captureSettings.withDataCaptureFileNameFormat(format);
+        return this;
+    }
+
+    @Override
+    public EventHubImpl withNewSendRule(final String ruleName) {
+        addPostRunDependent(context -> manager().eventHubAuthorizationRules()
+            .define(ruleName)
+            .withExistingEventHub(ancestor().resourceGroupName(), ancestor().ancestor1Name(), name())
+            .withSendAccess()
+            .createAsync()
+            .last());
+        return this;
+    }
+
+    @Override
+    public EventHubImpl withNewListenRule(final String ruleName) {
+        addPostRunDependent(context -> manager().eventHubAuthorizationRules()
+            .define(ruleName)
+            .withExistingEventHub(ancestor().resourceGroupName(), ancestor().ancestor1Name(), name())
+            .withListenAccess()
+            .createAsync()
+            .last());
+        return this;
+    }
```

Comment:
Please try test case which do both in one Update. I remember service not able to handle concurrent requests under eventhub, so these had to be done in sequence.

Reply:
Do they have `without` method?

Reply:
I will add new method `withNewSendAndListenRule(ruleName)`. There is `withoutAuthorizationRule(ruleName)` to remove the authorization rule no matter what access it has.

Reply:
Eh, I didn't get it. What I mean is these requests under "eventhub" might need to be called sequentially, it might not be limited to this 2.

Reply:
After discuss offline, we decide to concat the post run tasks in eventhubs.